### Задание 1. Базовое изучение

Изучить представленный набор данных на основе [описания его столбцов](https://raw.githubusercontent.com/obulygin/pyda_homeworks/master/statistics_basics/horse_data.names) и выбрать 8 столбцов для дальнейшего изучения (среди них должны быть как числовые, так и категориальные). Провести расчет базовых метрик для них, кратко описать результаты.

In [1]:
import pandas
from HorsesService import HorsesService

In [2]:
horsesService = HorsesService()
horse_data = horsesService.fetch_processed_hourse_data_frame()

In [3]:
horse_data.median()

is_surgery                          1.00
age                                 1.00
hospital_number                530305.50
rectal_temperature                 38.20
pulse                              64.00
respiratory_rate                   24.50
temperature_of_extremities          3.00
peripheral_pulse                    2.00
mucous_membranes                    3.00
capillary_refill_time               1.00
pain                                3.00
peristalsis                         3.00
abdominal_distension                2.00
nasogastric_tube                    2.00
nasogastric_reflux                  1.00
nasogastric_reflux_ph               5.00
rectal_examination_-_feces          3.00
abdomen                             4.00
packed_cell_volume                 45.00
total_protein                       7.50
abdominocentesis_appearance         2.00
abdomcentesis_total_protein         2.25
outcome                             1.00
is_surgical_lesion                  1.00
type_of_lesion_1

In [4]:
horse_data.describe()

age  hospital_number  rectal_temperature       pulse  \
count  300.000000     3.000000e+02          240.000000  276.000000   
mean     1.640000     1.085889e+06           38.167917   71.913043   
std      2.173972     1.529801e+06            0.732289   28.630557   
min      1.000000     5.184760e+05           35.400000   30.000000   
25%      1.000000     5.289040e+05           37.800000   48.000000   
50%      1.000000     5.303055e+05           38.200000   64.000000   
75%      1.000000     5.347275e+05           38.500000   88.000000   
max      9.000000     5.305629e+06           40.800000  184.000000   

       respiratory_rate  temperature_of_extremities  peripheral_pulse  \
count        242.000000                  244.000000        231.000000   
mean          30.417355                    2.348361          2.017316   
std           17.642231                    1.045054          1.042428   
min            8.000000                    1.000000          1.000000   
25%           18.500000                    1.000000          1.000000   
50%           24.500000                    3.000000          2.000000   
75%           36.000000                    3.000000          3.000000   
max           96.000000                    4.000000          4.000000   

       capillary_refill_time  packed_cell_volume  total_protein  \
count             268.000000          271.000000     267.000000   
mean                1.305970           46.295203      24.456929   
std                 0.477629           10.419335      27.475009   
min                 1.000000           23.000000       3.300000   
25%                 1.000000           38.000000       6.500000   
50%                 1.000000           45.000000       7.500000   
75%                 2.000000           52.000000      57.000000   
max                 3.000000           75.000000      89.000000   

       abdomcentesis_total_protein  
count                   102.000000  
mean                      3.019608  
std                       1.968567  
min                       0.100000  
25%                       2.000000  
50%                       2.250000  
75%                       3.900000  
max                      10.100000

In [5]:
hospital_number_percent = len(horse_data['hospital_number'].drop_duplicates()) / len(horse_data['hospital_number'])
print('Переболели единожды {:.0%}'.format(hospital_number_percent))

surgery_positively_percent =  len(horse_data[horse_data['is_surgery'] == True]) / len(horse_data['is_surgery'])
print('Перенесли операцию: {:.0%}'.format(surgery_positively_percent))

outcome_percent = (len(horse_data['outcome']) - len(horse_data[horse_data['outcome'] == '1'])) / len(horse_data['outcome'])
print('Умерли: {:.0%}'.format(outcome_percent))

outcome_percent = len(horse_data[horse_data['cp_data'] == True]) / len(horse_data['cp_data'])
print('Патологии для данной болезни: {:.0%}'.format(outcome_percent))

Переболели единожды 95%
Перенесли операцию: 60%
Умерли: 41%
Патологии для данной болезни: 33%


### Сводка по данным

- Возраст: взрослая лошадь
- Перенесли операцию: 60%
- Вылечили без операции: 40%
- Номера лошадей говорит что 95% переболели единожды
- Пульс здоровой лошади 30-40, медиана больных 64
- Частота дыхания здоровой лошади 8-10, медиана больных лошадей 24
- Температура конечностей больных лошадей: холодная (cool)
- Периферический пульс: вырос (increased)
- Слизистые оболочки: бледно-розовый (pale pink)
- Время наполнения капилляров меньше 3 сек, что хорошо
- Ощущение боли: прерывистая легкая боль (intermittent mild pain)
- Перистальтика: hypomotile
- Вздутие живота: слабый (slight)
- Газы из назогастрального зонда: слабый (slight)
- Рефлюкс из назогастрального зонда: none
- HP Рефлюкс из назогастрального зонда: чуть выше нормы
- Калл: уменьшен (decreased)
- У больных лошадей растянутый тонкий кишечник, что говорит о необходимости хирургического вмешательства
- Кровь (packed cell volume) в нормальном диапазоне
- Общий белок находится на верхней границе нормального диапазона
- Жидкость из брюшной полости: cloudy
- Уровень протеина: 2.25
- Умерли: 41%
- Патологии для данной болезни: 33%

### Общий портрет болеющий лошади:
Это взрослая лошадь, чаще требовалась операция из-за изменившегося размера кишечника (растянутый тонкий).

Болезнь можно увидеть без сложных медицинских приборов и исследований:
 - есть легкое вздутие живота
 - учащенное дыхание
 - холодные конечности
 - уменьшен калл

Почти все лошади болеют единожды. Большинство лошадей болеют без патологий (70%).
Данная болезнь несет за собой высокую смертность 41%.